### DISCUSSION TITLE
BLE-MI Compilation Error

### DISCUSSION THREAD

**Discussion by kskenyon** [arendst/Tasmota#20214] [Support request]
/Users/private/Tasmota/tasmota/tasmota_xsns_sensor/xsns_62_esp32_mi.ino:106:52: error: expected class-name before '{' token
 class MI32AdvCallbacks: public NimBLEScanCallbacks {
                                                    ^
/Users/private/Tasmota/tasmota/tasmota_xsns_sensor/xsns_62_esp32_mi.ino: In function 'void MI32ScanTask(void*)':
/Users/private/Tasmota/tasmota/tasmota_xsns_sensor/xsns_62_esp32_mi.ino:1067:13: error: 'class NimBLEScan' has no member named 'setScanCallbacks'
   MI32Scan->setScanCallbacks(&MI32ScanCallbacks,true);
             ^~~~~~~~~~~~~~~~
/Users/private/Tasmota/tasmota/tasmota_xsns_sensor/xsns_62_esp32_mi.ino: In function 'void MI32ConnectionGetCharacteristics(NimBLERemoteService*)':
/Users/private/Tasmota/tasmota/tasmota_xsns_sensor/xsns_62_esp32_mi.ino:1190:35: error: 'class NimBLERemoteCharacteristic' has no member named 'getProperties'
     MI32.conCtx->buffer[i] = chr->getProperties(); // flags as bitfield
                                   ^~~~~~~~~~~~~

### PROBLEM DESCRIPTION
_A clear and concise description of what the problem is._

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [ ] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [ ] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [ ] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [ ] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [ ] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [ ] Device used (e.g., Sonoff Basic): _____
- [ ] Tasmota binary firmware version number used: _____
  - [ ] Pre-compiled
  - [ ] Self-compiled
- [ ] Flashing tools used: _____
- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:


```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:


```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
_Steps to reproduce the behavior:_

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by Staars** [comment]
You have somehow created a build environment, that tries to compile the legacy MI32 driver with the `NimBLE-Arduino` library. This can not work, as this driver needs `esp-nimble-cpp`.

So you need to put the unwanted Bluetooth framework to `lib_ignore` in your personal custom build environment, depending on the BLE driver you want to build.
